src/page-modules/departures/server/journey-planner/index.ts

```diff
@@ -166,6 +166,21 @@ export function createJourneyApi(
         throw validated.error;
       }
 
+      validated.data.quays.sort((a, b) => {
+        // Place quays with no departures at the end
+        if (a.departures.length === 0) return 1;
+        if (b.departures.length === 0) return -1;
+
+        if (!a.publicCode) return 1;
+        if (!b.publicCode) return -1;
+
+        if (parseInt(a.publicCode) && parseInt(b.publicCode)) {
+          return parseInt(a.publicCode) - parseInt(b.publicCode);
+        }
```

Comment:
Maybe these can be combined to parse and check value as an integrated step?

Reply:
I'm not sure I understand 😅 Are you thinking of avoiding doing `parseInt` twice per variable? 🤔

Reply:
Also, do you know what type of mechanism is good to check edge cases for things like this? 😅

Reply:
.. I can write some tests 😄